mysys/my_getopt.c

```diff
@@ -981,6 +981,12 @@ static inline long eval_num_suffix(char *suffix, int *error)
     num*= 1024L * 1024L;
   else if (*suffix == 'g' || *suffix == 'G')
     num*= 1024L * 1024L * 1024L;
+  else if (*suffix == 't' || *suffix == 'T')
+    num*= 1024L * 1024L * 1024L * 1024L;
```

Comment:
Could you get rid of ```num``` and return values immediately instead?

Reply:
also L is the wrong suffix, as long is not capable to hold values over 4GB, cross-plattform. should be ULL or something

Reply:
I wonder if suffix is required here at all?

Reply:
#include <stdio.h>
int main()
{
   long long res= 1;
   res *= 1024*1024*1024*1024;
   printf("%lld\n", res);
   return 0;
}

this prints 0 on Windows, I just tried . If I change it to 
res *= 1024ULL*1024*1024*1024;
it printf a large number

So I suppose without suffix, the right side of the assignment is int, and as int it is 0, and then
res *= 0 gives 0.

Reply:
/me needs to study type propagation carefully.

An alternative would be res<<= 40;